contentcuration/automation/utils/appnexus/base.py

```diff
@@ -1,46 +1,183 @@
+        self.session.mount("http://", adapter)
+
+    def _construct_full_url(self, path):
+        """This method combine base_url, url_prefix, and path in that order, removing any trailing and leading slashes."""
+        url_array = []
+        if self.base_url:
+            url_array.append(self.base_url.rstrip("/"))
+        if self.url_prefix:
+            url_array.append(self.url_prefix.rstrip("/").lstrip("/"))
+        if path:
+            url_array.append(path.lstrip("/"))
+        return "/".join(url_array)
+
+    def _make_request(self, request):
+        url = self._construct_full_url(request.path)
+        try:
+            return self.session.request(
+                request.method,
+                url,
+                params=request.params,
+                data=request.data,
+                headers=request.headers,
+                json=request.json,
+                timeout=request.timeout,
+            )
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.RequestException,
+            requests.exceptions.SSLError,
+        ) as e:
+            logging.exception(e)
+            raise errors.ConnectionError(f"Unable to connect to {url}")
+        except (
+            requests.exceptions.Timeout,
+            requests.exceptions.ConnectTimeout,
+            requests.exceptions.ReadTimeout,
+        ) as e:
+            logging.exception(e)
+            raise errors.TimeoutError(f"Timeout occurred while connecting to {url}")
+        except (
+            requests.exceptions.TooManyRedirects,
+            requests.exceptions.HTTPError,
+        ) as e:
+            logging.exception(e)
+            raise errors.HttpError(f"HTTP error occurred while connecting to {url}")
+        except (
+            requests.exceptions.URLRequired,
+            requests.exceptions.MissingSchema,
+            requests.exceptions.InvalidSchema,
+            requests.exceptions.InvalidURL,
+            requests.exceptions.InvalidHeader,
+            requests.exceptions.InvalidJSONError,
+        ) as e:
+            logging.exception(e)
+            raise errors.InvalidRequest(f"Invalid request to {url}")
+        except (
+            requests.exceptions.ContentDecodingError,
+            requests.exceptions.ChunkedEncodingError,
+        ) as e:
+            logging.exception(e)
+            raise errors.InvalidResponse(f"Invalid response from {url}")
+    
     @abstractmethod
-    def connect(self) -> None:
+    def connect(self, **kwargs):
         """ Establishes a connection to the backend service. """
-        pass
+        try:
+            request = BackendRequest(method="GET", path=self.connect_endpoint, **kwargs)
+            response = self._make_request(request)
+            if response.status_code != 200:
+                return False
+            return True
+        except Exception as e:
+            return False
 
     @abstractmethod
```

Comment:
Same here-- no longer an `@abstractmethod`?

Reply:
Thank you, removed!